core/execution/engine_snapshot_test.go

```diff
@@ -343,8 +343,9 @@ func TestValidSettledMarketSnapshot(t *testing.T) {
 	marketConfig := getMarketConfig()
 	err := engine.SubmitMarket(ctx, marketConfig, "", time.Now())
 	assert.NoError(t, err)
-	// fake settled market
-	marketConfig.State = types.MarketStateSettled
+	// this does not work. Setting the state to settled will update the state on the market object
+	// unfortunately, it does not set the closed flag, so OnTick will not work
+	// marketConfig.State = types.MarketStateSettled
 	engine.OnTick(ctx, time.Now())
```

Comment:
What is `TestValidSettledMarketSnapshot` actually testing then if its not testing the snapshot of a settled market? Maybe the test should be changed to actually make a settled market properly if it wasn't working before.

Reply:
Yeah, so there were 2 failing tests, this one is indeed about specifically settled markets. I'll see if I can update the test and properly settle the market in this test